### PR TITLE
Fixed scp-style URIs with :/ in their paths

### PR DESCRIPTION
**Pull request by zkat**

### CODE REVIEW

File: normalize-git-url.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 var url = require('url')
 
 module.exports = function normalize (u) {
-  var parsed = url.parse(u, true)
+  var parsed = url.parse(u)
+  // If parsing actually alterns the URL, it is almost certainly a suspicious
```
Comment (othiym23) [inline normalize-git-url.js]: typo: `s/alterns/alters/`

Comment (othiym23) [inline normalize-git-url.js]: (also, what does "suspicious" mean here?)

File: test/basic.js [hunk 2]
```diff
@@ -24,6 +24,10 @@ test('basic normalization tests', function (t) {
     { url: 'git@github.com:npm/npm.git', branch: 'v1.0.27' }
   )
   t.same(
+    normalize('git+ssh://git@github.com:/npm/npm.git#v1.0.28'),
```
Comment (othiym23) [inline test/basic.js:27]: ugh... do we really want to treat this URL as valid?

Comment (zkat) [inline test/basic.js:27]: Absolutely! scp-style syntax requires the semicolon. That's the only way for it to know that an absolute URL is remote:

`scp git@myhost/x/y => ~error~`

That said, `ssh://` _is_ able to make ssh connections without the scp-style syntax.

Comment (othiym23) [inline test/basic.js:27]: _heavy sigh_ ookayyyyyyyeeeeee

In that case, LGTM. :sheep:

> Absolutely!

iswydt

**Pull request closed without merging**